File: nodejs/src/advent.js
const day = process.argv[2];

if (!day) {
  console.error("Please provide a number as an argument");
  process.exit(1);
}

const adventFunction = require(`./day${day}.js`);

const before = Date.now();
adventFunction();
const after = Date.now();

// time in seconds
const inSeconds = (after - before) / 1000;
console.log(`${inSeconds}s [JavaScript]`);

File: ruby/advent.rb
# get param from command line
day = ARGV[0]

if day.nil?
  puts "Please provide a number as an argument"
  exit
end

require_relative "day#{day}.rb"
advent = method("day#{day}")

start = Time.now
advent.call
finish = Time.now

puts "#{(finish - start).round(3)}s [Ruby]"
